src/System.Data.Common/src/System/Data/SQLTypes/SQLDecimal.cs

```diff
@@ -2267,7 +2267,15 @@ internal void AdjustScale(int digits, bool fRound)
                 throw new OverflowException(SQLResource.ArithOverflowMessage);
 
             bNewScale = (byte)(lAdjust + _bScale);
-            bNewPrec = (byte)(Math.Min(s_NUMERIC_MAX_PRECISION, Math.Max(1, lAdjust + _bPrec)));
+
+            if (Value == Decimal.Zero && _bScale == 0 && lAdjust > 0)
+            {
+                bNewPrec = bNewScale;
```

Comment:
Are you guaranteed to have Precision==1 in this case? Should you check for that in the if case, or at least add a debug assert for it? Do you also need to preserve the checks for negative precision and max precision like in the original line?